tools/engine_tool/lib/src/commands/run_command.dart

```diff
@@ -0,0 +1,141 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' show ProcessStartMode;
+
+import 'package:engine_build_configs/engine_build_configs.dart';
+import 'package:process_runner/process_runner.dart';
+
+import '../build_utils.dart';
+import 'command.dart';
+import 'flags.dart';
+
+/// The root 'run' command.
+final class RunCommand extends CommandBase {
+  /// Constructs the 'run' command.
+  RunCommand({
+    required super.environment,
+    required Map<String, BuilderConfig> configs,
+  }) {
+    builds = runnableBuilds(environment, configs);
+    debugCheckBuilds(builds);
+    // Add options here that are common to all queries.
+    argParser.addOption(
+      configFlag,
+      abbr: 'c',
+      defaultsTo: '',
+      help:
+          'Specify the build config to use for the target build (usually auto detected)',
+      allowed: <String>[
+        for (final Build build in runnableBuilds(environment, configs))
+          build.name,
+      ],
+      allowedHelp: <String, String>{
+        for (final Build build in runnableBuilds(environment, configs))
+          build.name: build.gn.join(' '),
+      },
+    );
+  }
+
+  /// List of compatible builds.
+  late final List<Build> builds;
+
+  @override
+  String get name => 'run';
+
+  @override
+  String get description => 'Invokes flutter run with a local engine build\n'
+      'All arguments after -- are forwarded to flutter run, e.g.: '
+      'et run -- --profile\n'
+      'See `flutter run --help` for a listing';
+
+  Build? _lookup(String configName) {
+    return builds.where((Build build) => build.name == configName).firstOrNull;
+  }
+
+  Build? _findHostBuild(Build? targetBuild) {
+    if (targetBuild == null) {
+      return null;
+    }
+
+    // TODO(johnmccutchan): This is brittle, it would be better if we encoded
+    // the host config name in the target config.
+    final String name = targetBuild.name;
+    if (name.contains('_debug')) {
```

Comment:
For some scenarios the target is a host build:

```suggestion
    if (name.startsWith('host_') {
      return name;
    } else if (name.contains('_debug')) {
```

Reply:
Done.